logging/change_to_local_ssd.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script sets the yarn.nodemanager.log-dirs value to the Local SSD mounts.
+
+# Dataproc configurations
+# Modify these configurations as needed
+# MAX_MNT_DISK_FOR_LOGS is the maximum number of disks to be used for creating the logs
+# LOGPATH is the path under /mnt/<mount number> where the application logs are created
+readonly MAX_MNT_DISK_FOR_LOGS=3
+readonly LOGPATH='hadoop/yarn/userlogs'
+
+# DO NOT CHANGE THESE VARIABLES UNLESS THE SOURCE FRAMEWORKS MODIFY IT
+readonly FLUENTD_YARN_USERLOGS='/etc/google-fluentd/config.d/dataproc-yarn-userlogs.conf'
+readonly HADOOP_CONF_DIR='/etc/hadoop/conf'
+
+function set_hadoop_property() {
+    local -r config_file=$1
+    local -r property=$2
+    local -r value=$3
+    bdconfig set_property \
+        --configuration_file "${HADOOP_CONF_DIR}/${config_file}" \
+        --name "${property}" --value "${value}" \
+        --clobber
+}
+
+# This function validates the MAX_MNT_DISK_FOR_LOGS variable and ensures that the value is capped at the max disks available
+function get_max_local_ssd() {
+    max_disk_no=0
+    for line in $(df -h --output=target); do
+        if [ ${line:0:4} = '/mnt' ]; then
```

Comment:
Logic dependent on hard coded values should be avoided. We should be able to get the localssd information from the cluster configuration

Reply:
As discussed, the dataproc metadata only exposes the types of disks and not the path itself. The hardcoding here is based off assumptions in GCE as well as Dataproc APIs